pkg/trafficpolicy/types.go

```diff
@@ -59,10 +59,11 @@ type InboundTrafficPolicy struct {
 	Rules     []*Rule  `json:"rules:omitempty"`
 }
 
-// Rule is a struct that represents which service identities (authenticated principals) can access a Route
+// Rule is a struct that represents which authenticated principals can access a Route.
+// A principal is of the form <service-identity>.<trust-domain>. It can also contain wildcards.
 type Rule struct {
 	Route                    RouteWeightedClusters `json:"route:omitempty"`
-	AllowedServiceIdentities mapset.Set            `json:"allowed_service_identities:omitempty"`
+	AllowedServiceIdentities mapset.Set            `json:"allowed_princinpals:omitempty"`
```

Comment:
only if we ever log it maybe? and even then i'm not sure if the logger we use attempts to marshal objects to json.

@shashankram do you know?

Reply:
We probably don't need to have JSON tags for internal structs in most cases, except when the type is used with the debugger pkg to render JSON. If these types are not used by the debugger pkg, we can remove them. In most cases, it exists because of following a pattern and for no apparent reason.